sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -243,6 +243,55 @@ case class MapGroups(
     outputObjAttr: Attribute,
     child: LogicalPlan) extends UnaryNode with ObjectProducer
 
+/** Factory for constructing new `FlatMapGroupsInR` nodes. */
+object FlatMapGroupsInR {
+  def apply(
+      func: Array[Byte],
+      packageNames: Array[Byte],
+      broadcastVars: Array[Broadcast[Object]],
+      schema: StructType,
+      encoder: Expression,
+      keyEncoder: Expression,
+      rowEncoder: ExpressionEncoder[Row],
```

Comment:
Actually how about just using the same names as constructor parameters of case class `FlatMapGroupsInR` below?